util: Remove useless formation indicator and add a missing unit test

Maledong opened the pull request:
Change Logs：

1) Remove format indicator (something like `${}`) because we don't need it here.
2) Add a type check because if the input value isn't of string, directly return `undefined` (We don't need to go through the `switch case` for check to save time).
3) Add a missing test for object type in `test-internal-util-normalizeencoding.js`.

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

Maledong commented:
@mscdex：At last I abandoned re-writing after many tests about the efficiency but I changed the title and submit to add a missing test for utils. Plz have a review. Thanks anyway!

apapirovski commented:
Lite CI: https://ci.nodejs.org/job/node-test-pull-request-lite/893/

@Maledong you might want to associate your commit email with your account as otherwise you won't get proper credit for contributing to the project from Github.

Maledong commented:
OK. Thanks @apapirovski :)

Because my previous email accout is somehow reguarded by GitHub as an invalid one, I removed that according to the GitHub's guides and re-create a new one. So now you can see this.

BTW：The original code has missed `'ucs-2'` check in different kind of cases (e.g: UcS-2). So I've added the missing codes in my submitted code. Plz have a review again, thanks anyway!

Maledong commented:
OK, I've reset my accout with a new one verified by GitHub. And everything goes well with me.
@apapirovski ：I see this test failed. Is there anything to be fixed with?

apapirovski commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/15607/

Maledong commented:
Is there anything to change? I see that it can pass all the tests except for on mac……

joyeecheung commented:
Jenkins failed to fetch the source code on macOS. Just to be safe, a new CI: https://ci.nodejs.org/job/node-test-pull-request/15632/

Maledong commented:
@joyeecheung：Many thanks!
If you have good suggestions for my changed codes. You can also have a review for that as well :)

Maledong commented:
OK，thanks anyway!

Maledong commented:
@joyeecheung & @apapirovski ：
Sorry to interrupt you both but I wanna say, since this has fixed a missing code conversion for `ucs-2` and others, and this has passed all the tests, so just a reminder to merge them into your Node master if you can.
Thanks anyway!

Trott commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/15759/

Trott commented:
Adding `author ready` so this doesn't get missed again if it's ready to land. Whoever lands it will need to edit the commit message. (`test & util` -> `test, util:`).

Trott commented:
Re-running Windows only: https://ci.nodejs.org/job/node-test-commit-windows-fanned/19156/

Maledong commented:
Thanks anyway.
